docs/VALIDATOR_TESTS.md

```diff
@@ -0,0 +1,194 @@
+
+## Background
+
+Every validator test is represented by a dedicated type (struct) which implements the `Validator` interface (`pkg/validate/validate_test.go`)
+
+```go
+type Validator interface {
+	Name() string
+	Run(utils.MetaBundle) (bool, error)
+	SucceedingCandidates() []utils.MetaBundle
+	FailingCandidates() []utils.MetaBundle
+}
+```
+
+For implementing that interface, the Validator test's struct must have four methods:
+- `Name() string` : Returns the name of the validator to which the test corresponds.
+- `SucceedingCandidates() []utils.MetaBundle`: Returns a slice of `MetaBundle` where each element can successfully pass the validator.
+- `FailingCandidates() []utils.MetaBundle`: Returns a slice of `MetaBundle` where each element fails the validator due to any reason.
+- `Run(utils.MetaBundle) (bool, error)`:  Represents running the validator over a `MetaBundle` and returning a `bool` (success: true/false) and `error` (error while executing the validator) accordingly. Succeeding Candidate `MetaBundle` fed to this method will be expected to return `(true, nil)` and Failing Candidate `MetaBundle` fed to this method will be expected to return `(false, nil/non-nil)`.
+
+**Checkout an [Example](#example)**
+
+
+## How to add a new validator test
+
+Say, you created a new validator called `validator_default_channel` under the file `pkg/validators/validator_default_channel.go` accordingly.
+
+* Now, go ahead and create a new file `pkg/validators/test_validator_default_channel.go`.
+* Inside this file, define a struct corresponding to this validator. It can be an empty struct or non-empty depending on the way you want to implement the testing of it. For example: `type ValidateDefaultChannelTestBundle struct {}`
+* Now, make this struct implement the `Validator` interface defined under `pkg/validate/validate_test.go`. (Refer to the previous section for the explanation for these methods).
+* After this, your tests are ready but you have to register them to the validators' test suite now.
+* For that, proceed to `pkg/validate/validate_test.go` and look for `var validatorsToTest`.
+* Under it, just append the struct corresponding to the validator-test you defined under `pkg/validators/test_validator_default_channel.go`. For example, in our case, it would look like this:
+```go
+...
+// register the validators to test here by appending them to the following slice
+var validatorsToTest []Validator = []Validator{
+    ValidatorAddonLabelTestBundle{},  // ignore this, it was previously present
+    ValidatorDefaultChannelTestBundle{}, // pay attention! we just added this
+}
+...
+```
+* That's it! Rest will be taken care of assuming that the methods you defined for the struct are aptly coded.
+* Go ahead and run the tests by entering `make test-validators` and it will run:
```

Comment:
would change this line `make test-validators` as this target is not ran in `pr_check.sh`